experimental/Schemes/FunctionFields.jl

```diff
@@ -0,0 +1,371 @@
+
+export VarietyFunctionFieldElem
+export representative
+
+
+### Check for irreducibility
+@attr Bool function is_irreducible(X::AbsCoveredScheme)
+  for U in basic_patches(default_covering(X))
+    is_irreducible(U) || return false
+  end
+  C = default_covering(X)
+  # Check that X is connected
+  for U in patches(C)
+    for V in patches(C) 
+      A, _ = glueing_domains(C[U, V])
+      is_empty(A) && return false
+    end
+  end
+  return true
+end
+
+
+########################################################################
+# lower checks for irreducibility                                      #
+########################################################################
+@attr Bool function is_irreducible(X::AbsSpec{<:Field, <:MPolyRing}) 
+  return true
+end
+
+@attr Bool function is_irreducible(X::AbsSpec{<:Ring, <:MPolyQuo}) 
+  return is_prime(modulus(OO(X)))
+end
+
+@attr Bool function is_irreducible(X::AbsSpec{<:Field, <:MPolyLocalizedRing}) 
+  return true
+end
+
+@attr Bool function is_irreducible(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing}) 
+  return is_prime(localized_modulus(OO(X)))
+end
+
+
+########################################################################
+# Check for emptyness                                                  #
+########################################################################
+
+@attr Bool function Base.isempty(U::SpecOpen)
+  return all(isempty, affine_patches(U))
+end
+
+########################################################################
+# Rational functions on irreducible varieties                          #
+########################################################################
+
+mutable struct VarietyFunctionField{BaseRingType<:Field, 
+                                    FracFieldType<:AbstractAlgebra.Generic.FracField,
+                                    CoveredSchemeType<:AbsCoveredScheme,
+                                    SpecType<:AbsSpec
+                                   } <: Field
+  kk::BaseRingType
+  X::CoveredSchemeType
+  U::SpecType  # representative patch to represent rational functions
+  KK::FracFieldType
+
+  function VarietyFunctionField(
+      X::AbsCoveredScheme; 
+      check::Bool=true,
+      representative_patch::AbsSpec=default_covering(X)[1]
+    )
+    check && (is_irreducible(X) || error("variety is not irreducible"))
+    representative_patch in default_covering(X) || error("representative patch not found")
+    KK = FractionField(ambient_ring(representative_patch))
+    kk = base_ring(X)
+    return new{typeof(kk), typeof(KK), typeof(X), typeof(representative_patch)}(kk, X, representative_patch, KK)
+  end
+end
+
+### essential getters
+representative_patch(KK::VarietyFunctionField) = KK.U
+variety(KK::VarietyFunctionField) = KK.X
+coefficient_ring(KK::VarietyFunctionField) = KK.kk
+representative_field(KK::VarietyFunctionField) = KK.KK
+
+### user facing constructors 
+function_field(X::CoveredScheme; 
```

Comment:
seems to ignore the arguments `check` and `representative_patch`
also `check` is always the last argument :-)